### PR TITLE
[MODULAR] Grants Hearthkin Transcendent Olfaction (& One Other Thing)

### PR DESCRIPTION
**Pull request by Nerev4r**
## About The Pull Request
This gives them that mutation that allows you to track people by scent.
I heavily recommend you have all the kindred nuzzle something at roundstart so you can track them later.
I was going to get a meme done for this PR body but this'll do instead.
image

This also gives them some brown bear pelts.

Thanks to @vinylspiders for helping me with the code.

## How This Contributes To The Skyrat Roleplay Experience
Easier allows them to track people for all kinds of purposes, and the brown bear pelt is kinda drip.

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>
  
image

</details>

## Changelog
:cl:
add: Hearthkin have discovered a new herbal remedy to clear the sinuses, and now enjoy bloodhound-level scent tracking as they did long ago.
add: Hearthkin have discovered a population of grizzly bears on the ice moon, and have appropriately skinned some of them.
/:cl:

### CODE REVIEW

**Comment by Nerev4r** [comment]
image

i'm just going to post every meme ppl make for this PR until it's merged

**Comment by SpringSkipper** [comment]
icecats aren't allowed to have snouts by server policy. This is FailRP. :neckbeard:
/veryserious

File: modular_skyrat/modules/primitive_catgirls/code/species.dm [hunk 1]
```diff
@@ -35,6 +35,25 @@
 
 	always_customizable = TRUE
 
+/datum/species/human/felinid/primitive/on_species_gain(mob/living/carbon/new_primitive, datum/species/old_species, pref_load)
+	. = ..()
+	var/mob/living/carbon/human/hearthkin = new_primitive
+	if(!istype(hearthkin))
+		return
+	hearthkin.dna.add_mutation(/datum/mutation/human/olfaction)
```
Comment (vinylspiders) [inline modular_skyrat/modules/primitive_catgirls/code/species.dm]:
```suggestion
	hearthkin.dna.add_mutation(/datum/mutation/human/olfaction, MUT_NORMAL)
```
Might fix the CI issues?

File: modular_skyrat/modules/primitive_catgirls/code/species.dm [hunk 2]
```diff
@@ -35,6 +35,25 @@
...
+	hearthkin.dna.activate_mutation(/datum/mutation/human/olfaction)
+    // >mfw I take mutadone and my nose clogs
```
Comment (vinylspiders) [inline modular_skyrat/modules/primitive_catgirls/code/species.dm]:
```suggestion
    	// >mfw I take mutadone and my nose clogs
```

**Review by vinylspiders — CHANGES_REQUESTED** [review]

**Comment by Ephemeralis** [comment]
I think it might be better to do this via `inert_mutation` on `/datum/species` and then activate it on species gain instead. This also has the added bonus of having the base `on_species_loss` proc handle swapping this out appropriately if an icewalker loses their species somehow.

**Comment by Ephemeralis** [comment]
Bizarre CI issues notiwithstanding, I've had some limited success with this implementation locally:

```dm

/datum/mutation/human/olfaction/primitive
	mutadone_proof = TRUE
	instability = 0

/datum/species/human/felinid/primitive/on_species_gain(mob/living/carbon/hearthkin, datum/species/old_species, pref_load)
	. = ..()
	hearthkin.dna.add_mutation(/datum/mutation/human/olfaction/primitive, MUT_EXTRA)

/datum/species/human/felinid/primitive/on_species_loss(mob/living/carbon/human/hearthkin, datum/species/new_species, pref_load)
	. = ..()
	// if we have icecat transcendent olfaction (and we should, but check to be sure), lose it
	if(hearthkin.dna.mutation_in_sequence(/datum/mutation/human/olfaction/primitive))
		hearthkin.dna.remove_mutation(/datum/mutation/human/olfaction/primitive)
```

Whether this'll fare better in CI, I can't really say, but we do have a few problems:

- Attempting to change the dna for mutadone immunity causes a `modify()` runtime, maybe due to species_gain being called before the power stuff is initiated? Not fully sure. It doesn't seem to break anything immediate.
- Transcendent olfaction itself seems to stop working almost immediately due to miasma/cleanliness in the icemoon atmos. If you tap olfaction *twice* in an unclean area, you're disgusted for a seriously long period of time and puke everywhere, making it even harder to use olfaction again.

**Comment by Stalkeros2** [comment]
> image

this image proves that they need muskets

**Comment by vinylspiders** [comment]
> Bizarre CI issues notiwithstanding, I've had some limited success with this implementation locally:
> 
> ```dm
> /datum/mutation/human/olfaction/primitive
> 	mutadone_proof = TRUE
> 	instability = 0
> 
> /datum/species/human/felinid/primitive/on_species_gain(mob/living/carbon/hearthkin, datum/species/old_species, pref_load)
> 	. = ..()
> 	hearthkin.dna.add_mutation(/datum/mutation/human/olfaction/primitive, MUT_EXTRA)
> 
> /datum/species/human/felinid/primitive/on_species_loss(mob/living/carbon/human/hearthkin, datum/species/new_species, pref_load)
> 	. = ..()
> 	// if we have icecat transcendent olfaction (and we should, but check to be sure), lose it
> 	if(hearthkin.dna.mutation_in_sequence(/datum/mutation/human/olfaction/primitive))
> 		hearthkin.dna.remove_mutation(/datum/mutation/human/olfaction/primitive)
> ```
> 
> Whether this'll fare better in CI, I can't really say, but we do have a few problems:
> 
>     * Attempting to change the dna for mutadone immunity causes a `modify()` runtime, maybe due to species_gain being called before the power stuff is initiated? Not fully sure. It doesn't seem to break anything immediate.
> 
>     * Transcendent olfaction itself seems to stop working almost immediately due to miasma/cleanliness in the icemoon atmos. If you tap olfaction _twice_ in an unclean area, you're disgusted for a seriously long period of time and puke everywhere, making it even harder to use olfaction again.

I have a fix for the modify() issue in the works upstream, it's a race condition with the timer that calls that proc basically calling it after the mob has been qdeleted. 

image

I am having trouble finding the source of the hard del though. It's really odd...

Wonder if the mutation itself is just bugged.

**Comment by vinylspiders** [comment]
Testing this revealed an upstream bug that we now have to wait for the PR I made to come to us, but it should fix the ci issues

**Review by vinylspiders — APPROVED** [review]
The fix from upstream has arrived, at last this should be good to go now.

**Pull request merged**